src/units/KM_Units.pas

```diff
@@ -307,6 +313,79 @@ function TKMUnitCitizen.FindHome: Boolean;
 end;
 
 
+procedure TKMSettledUnit.CleanHousePointer(aFreeAndNilTask: Boolean = False);

```

Comment:
I don't like the name of this function. Maybe ResetHome would be better. I also think you should remove aFreeAndNilTask. That's unrelated to resetting the unit's home. It's only one extra line to call FreeAndNil(fUnitTask) and it's much more obvious what you are doing than a mysterious "true" parameter.